packages/expo-font/plugin/src/utils.ts

```diff
@@ -18,3 +18,56 @@ export async function resolveFontPaths(fonts: string[], projectRoot: string) {
       (p) => p.endsWith('.ttf') || p.endsWith('.otf') || p.endsWith('.woff') || p.endsWith('.woff2')
     );
 }
+
+const weightMap: Record<string, number> = {
+  thin: 100,
+  extralight: 200,
+  ultralight: 200,
+  light: 300,
+  regular: 400,
+  normal: 400,
+  book: 400,
+  medium: 500,
+  semibold: 600,
+  demibold: 600,
+  bold: 700,
+  extrabold: 800,
+  ultrabold: 800,
+  black: 900,
+  heavy: 900,
+};
+const weights = Object.keys(weightMap).sort((a, b) => b.length - a.length);
+
+export function getFontWeight(filename: string) {
+  for (const weight of weights) {
+    if (filename.replaceAll('_', '').includes(weight)) {
+      return weightMap[weight];
+    }
+  }
+
+  return 400;
+}
+
+export function normalizeFilename(filename: string): string {
+  return filename
+    .toLowerCase()
+    .replace(/[\s-]+/g, '_')
+    .replace(/[^a-z0-9_]/g, '')
+    .replace(/_+/g, '_')
+    .replace(/^_+|_+$/g, '');
+}
+
+export function generateFontFamilyXml(files: string[]) {
+  let xml = `<?xml version="1.0" encoding="utf-8"?>\n<font-family xmlns:app="http://schemas.android.com/apk/res-auto">\n`;
+
+  files.forEach((file) => {
+    const filename = normalizeFilename(path.basename(file, path.extname(file)));
+    const fontWeight = getFontWeight(filename);
+    const fontStyle = filename.includes('italic') ? 'italic' : 'normal';
```

Comment:
instead of using heuristics to determine fontWeight and fontStyle, we should have the user provide these values (they both appear to be optional values - android studio says: `If unspecified, the value in the font's header tables will be used.`)